sources/src/main/resources/META-INF/resources/index.html

```diff
@@ -3,1007 +3,729 @@
+    constructor(selector) {
+      super(selector);
+
+      this.statusCaptions = {
+        "ACTIVE": "Active",
+        "INACTIVE": "Available",
+        "ACTIVATION_PENDING": "Waiting for approval"
+      };
+
+      this.activationTypeCaptions = {
+        "SELF_APPROVAL": "-",
+        "PEER_APPROVAL": "Peer approval required",
+        "EXTERNAL_APPROVAL": "External approval required"
+      };
+
+      this.activationTypes = [
+        "SELF_APPROVAL",
+        "PEER_APPROVAL",
+        "EXTERNAL_APPROVAL"
+      ];
+    }
+  }
+
+  class ListPrivilegesView extends PrivilegesViewBase {
+    constructor(model) {
+      super('#jit-listprivileges-view');
+
+      console.assert(model);
+
+      this._table = new mdc.dataTable.MDCDataTable(document.querySelector('#jit-privileges-table'));
+      this.model = model;
+    }
+
+    async showAsync() {
+      console.assert(document.appbar.scope);
+      super.showAsync();
+
+
+      let selectedPrivileges = []
+
+      try {
+        this._table.showProgress();
+        this._table.clearRows();
+
+        //
+        // Load privileges.
+        //
+        const privileges = await this.model.listRoles(document.appbar.scope);
+
+        if (privileges.roles && privileges.roles.length > 0) {
+          privileges.roles.forEach(item => {
+            this._table.addRow(
+              item.roleBinding.role,
+              [
+                { text: item.roleBinding.role },
+                { text: this.activationTypeCaptions[this.activationTypes.filter(type => item.activationType.startsWith(type))[0]] ?? item.activationType },
+                {
+                  text: (this.statusCaptions[item.status] ?? item.status) +
+                    (item.validUntil == null ? "" : ` (${Math.floor((item.validUntil * 1000 - new Date()) / (60 * 1000))} minutes left)`),
+                  class: item.status === "ACTIVE" ? "activated" : null
                 }
+              ],
+              true);
+          });
 
-                this.rowIndex += 1;
-            }
+          if (privileges.warnings && privileges.warnings > 0) {
+            document.appbar.showError(privileges.warnings.join(", "));
+          }
+        }
 
-            /** Get values of selected rows */
-            get selected() {
-                var boxes = $(`${this.selector} input:checked`).not(".checkall__input");
-                return $.map(boxes, input => input.value);
-            }
+        //
+        // Update controls when selection changes.
+        //
+        const onSelectionChanged = e => {
+          const selected = this._table.getSelectedRowIds();
+          selectedPrivileges = privileges.roles.filter(i => selected.includes(i.roleBinding.role));
+
+          if (selectedPrivileges.length > 1 && selectedPrivileges.some(i => i.activationType != "SELF_APPROVAL")) {
+            $('#jit-listprivileges-warning').text('You can only request multi-party approval for a single role at a time');
```

Comment:
Is it confusing that it says "multi-party" approval here, when we also have the "external approval" option or is it fine?

Reply:
I hope not. I deliberately changed it from `peer approval` to `multi-party approval` to better fit both types of approval.